### PR TITLE
Issue #11865: Resolve RegExp idea violations

### PR DESCRIPTION
**Pull request by Vyom-Yadav**
#11865

### CODE REVIEW

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/InputPackageNameSimple.java [hunk 1]
```diff
@@ -1,6 +1,6 @@
 /*
 PackageName
-format = (default)^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$
+format = (default)^^[a-z]+(\.[a-zA-Z_]\w*)*$
```
Comment (nick-mancuso) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/InputPackageNameSimple.java:3]: Why did we add extra `^` here?

**Review by nick-mancuso — CHANGES_REQUESTED** [review]
It looks like we can also catch some low-hanging fruit elsewhere with this update:

```bash
➜  checkstyle git:(master) ✗ ag -Q "[a-zA-Z0-9_]"
src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/InputPackageNameSimple.java
3:format = (default)^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$

src/main/java/com/puppycrawl/tools/checkstyle/checks/naming/PackageNameCheck.java
47: * Default value is {@code "^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$"}.
59: * package COM; // violation, name 'COM' must match pattern '^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$'

src/main/resources/com/puppycrawl/tools/checkstyle/meta/checks/naming/PackageNameCheck.xml
20:            <property default-value="^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$"

src/main/resources/google_checks.xml
351:      <property name="format" value="^[a-z][a-z0-9][a-zA-Z0-9_]*$"/>

src/it/java/com/google/checkstyle/test/chapter5naming/rule523methodnames/MethodNameTest.java
40:        final String format = "^[a-z][a-z0-9][a-zA-Z0-9_]*$";

src/xdocs/config_naming.xml
2280:              <td><code>"^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$"</code></td>
2297:package COM; // violation, name 'COM' must match pattern '^[a-z]+(\.[a-zA-Z_][a-zA-Z0-9_]*)*$'
➜  checkstyle git:(master) ✗ ag -Q "[ ]"         
...

src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java
247:            Pattern.compile("\n[ ]+(\\*)|^[ ]+(\\*)");

src/it/java/org/checkstyle/base/AbstractItModuleTestSupport.java
76:            .createPattern(".*[ ]*//[ ]*warn[ ]*|/[*]\\*?\\s?warn\\s?[*]/");


```
Please fix issues that inspections missed and add to this PR.

Item:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/InputPackageNameSimple.java (hunk 1 above)
Comment (Vyom-Yadav) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/naming/packagename/InputPackageNameSimple.java]: Removed.

**Comment by Vyom-Yadav** [comment]
> Please fix issues that inspections missed and add to this PR.

Done.

File: src/it/java/com/google/checkstyle/test/chapter5naming/rule523methodnames/MethodNameTest.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ protected String getPackageLocation() {
     public void testMethodName() throws Exception {
         final Configuration checkConfig = getModuleConfig("MethodName");
         final String msgKey = "name.invalidPattern";
-        final String format = "^[a-z][a-z0-9][a-zA-Z0-9_]*$";
+        final String format = "^[a-z][a-z0-9]\\w*$";
```
Comment (romani) [inline src/it/java/com/google/checkstyle/test/chapter5naming/rule523methodnames/MethodNameTest.java:40]: Is `_` is a `\w` ?

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: src/it/java/com/google/checkstyle/test/chapter5naming/rule523methodnames/MethodNameTest.java (hunk 2 above)
Comment (rnveach) [reply src/it/java/com/google/checkstyle/test/chapter5naming/rule523methodnames/MethodNameTest.java:40]: Yes. https://www3.ntu.edu.sg/home/ehchua/programming/howto/Regexe.html
> \w, \W: ANY ONE word/non-word character. For ASCII, word characters are [a-zA-Z0-9_]

Comment (romani) [reply src/it/java/com/google/checkstyle/test/chapter5naming/rule523methodnames/MethodNameTest.java:40]: Surprised, but good to know

**Review by romani — APPROVED** [review]
Ok to merge.
It ok to change default value, it is identical.

**Review by nick-mancuso — APPROVED** [review]

**Pull request merged**